prototype2/multiblade/mbcommon/MBConfig.cpp

```diff
@@ -0,0 +1,91 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief using nlohmann json parser to read configurations from file
+//===----------------------------------------------------------------------===//
+
+#include <common/Log.h>
+#include <fstream>
+#include <multiblade/mbcommon/MBConfig.h>
+#include <nlohmann/json.hpp>
+
+///
+MBConfig::MBConfig(std::string jsonfile) : ConfigFile(jsonfile) {
+
+  loadConfigFile();
+
+  if (instrument == InstrumentGeometry::Estia) {
+    detector = new MB16Detector(Digitisers);
+  }
+
+  assert(detector != nullptr);
+}
+
+///
+void MBConfig::loadConfigFile() {
```

Comment:
So, I guess the general idea here is to catch all exceptions and return from the function no matter what, then have the boolean variable indicate as to whether a valid config was loaded? That may be ok, but do you actually really want to run the pipeline without loading a valid (or default) config?

Reply:
No, this was the original idea, but I have reverted on using default values.
The idea is now just to catch all exceptions and provide hints to where the problem is. Maybe
I can throw a runtime exception if loading fails.